### PR TITLE
Fixes part of #40 & #42: Highfi InputInteraction 

### LINKED ISSUE #1644: Full Highfi for Ratio Interaction
This issue is tracking the full highfi work of ratio interaction under following circumstances:

1. Mobile Portrait
2. Mobile Portrait Split View
3. Mobile Landscape
4. Mobile Landscape Split View
5. Tablet Portrait
6. Tablet Portrait Split View
7. Tablet Landscape
8. Tablet Landscape Split View

**Note**: This issue is assigned to me considering two major points:
1. We don't have mocks for image region selection but in real we don't need those mocks because the general design for this interaction is similar to DragAndDropInteraction, TextInputInteraction, etc.
2. Implementing the highfi work in Question/Exploration Player needs context from other implementations too which I am already aware of.

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

Fixes part of #40
Fixes part of #42
Fix #1644

## Major changes
- Removes all other layout files
- Introduces styles which can be used for all input related interactions.

## Inner Margins Resources
- All possible layouts have same inner margins. [Link](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/16624e06-57b5-4d0a-a35e-ed2b06154cba)

## Outer Margins Resources
[Exploration Mobile Portrait](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/12d5b53e-8058-4b3c-ac95-fe53c07fcca6/)
[Exploration Mobile Landscape](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/9fa4204e-9936-4e35-9ca7-c1ecaee4ea7f)
[Exploration Tablet Portrait](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/dd6204c6-faf7-49b5-bd90-e87537f00715/)
[Exploration Tablet Landscape](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/25f15fb8-48ef-4e9a-bda5-d0b99648cf3a)
[Question Mobile Portrait](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/3ebedceb-6911-4260-ab08-5e3247aaa8c3)
[Question Mobile Landscape](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/208ac665-ca91-48c9-87f6-e7aa7ed3ac29)
[Question Tablet Portrait](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/723e0e8e-0da5-4613-88c3-4e155b31e289)
[Question Tablet Landscape](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/bfe0e299-2e0e-48c0-a74c-0b573d44117f)

## Note for Reviewer
The blue shadow based background does not look/correct as it results in mis-alignment of the interaction with the error message (which you can see in below screenshot) therefore the blue shadow needs to be changed but this will be discussed with Chantel first done later in a separate PR. (Filed #3178)

  

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/values-land/dimens.xml [hunk 1]
```diff
@@ -135,4 +129,14 @@
...
   <dimen name="feedback_item_question_view_margin_bottom">8dp</dimen>
+
+  <!-- Interaction Item: Exploration View -->
+  <dimen name="interaction_item_exploration_view_margin_start">80dp</dimen>
+  <dimen name="interaction_item_exploration_view_margin_top">24dp</dimen>
+  <dimen name="interaction_item_exploration_view_margin_end">48dp</dimen>
+
+  <!-- Interaction Item: Question View -->
+  <dimen name="interaction_item_question_view_margin_start">64dp</dimen>
+  <dimen name="interaction_item_question_view_margin_top">24dp</dimen>
```
Comment (anandwana001) [inline app/src/main/res/values-land/dimens.xml:138]: confirming this 32 or 24 or it's like 24+8 = 32?
https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/208ac665-ca91-48c9-87f6-e7aa7ed3ac29/

File: app/src/main/res/layout/text_input_interaction_item.xml [hunk 2]
```diff
@@ -12,36 +12,19 @@
...
+    app:layoutMarginStart="@{viewModel.hasConversationView ? @dimen/interaction_item_exploration_view_margin_start : @dimen/interaction_item_question_view_margin_start}"
+    app:layoutMarginTop="@{viewModel.hasConversationView ? @dimen/interaction_item_exploration_view_margin_top : @dimen/interaction_item_question_view_margin_top}">
 
     <org.oppia.android.app.customview.interaction.TextInputInteractionView
       android:id="@+id/text_input_interaction_view"
-      android:layout_width="match_parent"
-      android:layout_height="wrap_content"
-      android:background="@drawable/edit_text_background"
+      style="@style/InputInteractionEditText"
       android:contentDescription="@{viewModel.hintText.length() == 0 ? @string/text_input_default_content_description : ``}"
```
Comment (anandwana001) [inline app/src/main/res/layout/text_input_interaction_item.xml:24]: will it be ok to keep the content description empty in the false case?

```xml
android:contentDescription="@{viewModel.hintText.length() == 0 ? 
@string/text_input_default_content_description : ``}"
```

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -264,6 +264,36 @@
     <item name="android:textSize">14sp</item>
     <item name="fontFamily">sans-serif-medium</item>
   </style>
+
+  <style name="InputInteractionEditText" parent="Widget.AppCompat.EditText">
+    <item name="android:layout_width">match_parent</item>
+    <item name="android:layout_height">wrap_content</item>
+    <item name="android:background">@drawable/edit_text_background</item>
```
Comment (anandwana001) [inline app/src/main/res/values/styles.xml:271]: nit fix if you want, we can arrange these items in alphabetical order.

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -264,6 +264,36 @@
...
+    <item name="android:fontFamily">sans-serif</item>
+    <item name="android:imeOptions">actionDone</item>
+    <item name="android:longClickable">false</item>
+    <item name="android:maxLength">200</item>
+    <item name="android:minHeight">48dp</item>
```
Comment (anandwana001) [inline app/src/main/res/values/styles.xml]: how about using from dimen resource

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @rt4914 

nit suggestions

File: app/src/main/res/values-land/dimens.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/values-land/dimens.xml:138]: It should be `24dp` only. The remaining `8dp` was adjusted as part of margin bottom in content view: https://github.com/oppia/oppia-android/blob/233df50e3460f926e08c264973acde788aff530c/app/src/main/res/values-land/dimens.xml#L105

File: app/src/main/res/layout/text_input_interaction_item.xml (hunk 2 above)
Comment (rt4914) [reply app/src/main/res/layout/text_input_interaction_item.xml:24]: Actually for now we will need to keep this but once I confirm with Ben that all input-interactions will have a description we can remove this.

File: app/src/main/res/values/styles.xml (hunk 3 above)
Comment (rt4914) [reply app/src/main/res/values/styles.xml:271]: I kept it in order of `layout/xml` files where `layout_width` and `layout_height` are always on top. and rest all are in alphabetical order.

File: app/src/main/res/values/styles.xml (hunk 4 above)
Comment (rt4914) [reply app/src/main/res/values/styles.xml]: Done.

**Review by anandwana001 — APPROVED** [review]
LGTM

**Pull request merged**